Tools/TestWebKitAPI/Tests/ios/WKScrollViewTests.mm

```diff
@@ -172,6 +172,30 @@ inline static UIScrollPhase legacyScrollPhase(WKBEScrollViewScrollUpdatePhase ph
 @interface WKWebView (WKBEScrollViewDelegate)
 - (void)scrollView:(UIScrollView *)scrollView handleScrollUpdate:(WKBEScrollViewScrollUpdate *)update completion:(void (^)(BOOL handled))completion;
 @end
+
+@interface WKWebView (WKScrollViewTests)
+- (BOOL)synchronouslyHandleScrollEventWithPhase:(WKBEScrollViewScrollUpdatePhase)phase location:(CGPoint)location delta:(CGVector)delta;
+@end
+
+@implementation WKWebView (WKScrollViewTests)
+
+- (BOOL)synchronouslyHandleScrollEventWithPhase:(WKBEScrollViewScrollUpdatePhase)phase location:(CGPoint)location delta:(CGVector)delta
+{
+    __block bool done = false;
+    __block bool scrollingPrevented;
+
+    RetainPtr update = createScrollUpdate(phase, location, delta);
+    [self scrollView:self.scrollView handleScrollUpdate:update.get() completion:^(BOOL prevented) {
+        scrollingPrevented = prevented;
```

Comment:
Implicit `BOOL` to `bool`, I suppose that's fine? I thought it could be problematic sometimes.

Reply:
Oh, right — I meant to use `BOOL` for the value here (although implicit conversion is technically okay in this case). AFAIK, the only case where the implicit conversion causes problems is on Catalyst, if the compiler tries to determine the implicit return type of a C++ lambda that has branches which return both a `bool` and a `BOOL`. A simple example of this is something like:

```
auto result = [&] {
    if (returnBool())
        return false;

    BOOL value = getValue();
    return value;
}();
```